src/rules/no-duplicate-store-ids.ts

```diff
@@ -0,0 +1,47 @@
+import { createEslintRule } from '../utils/rule-creator'
+import { AST_NODE_TYPES } from '@typescript-eslint/utils'
+
+export const RULE_NAME = 'no-duplicate-store-ids'
+export type MESSAGE_IDS = 'duplicatedStoreIds'
+type Options = []
+
+const usingStoreIds = new Set<string>()
+
+export default createEslintRule<Options, MESSAGE_IDS>({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow duplicate store ids.'
+    },
+    schema: [],
+    messages: {
+      duplicatedStoreIds: 'No duplicated store ids allowed.'
+    }
+  },
+  defaultOptions: [],
+  create: (context) => {
+    return {
+      CallExpression(node) {
+        const callee = node.callee
+        if (callee.type !== 'Identifier' || callee.name !== 'defineStore')
+          return
+
+        const storeId = node.arguments && node.arguments[0]
+
+        if (!storeId || storeId.type !== AST_NODE_TYPES.Literal) return
+
+        const value = storeId.value as string
+
+        if (usingStoreIds.has(value)) {
+          context.report({
+            node,
```

Comment:
We can report on the actual literal node to make it more clear.

```suggestion
            node: storeId,
```